src/Microsoft.ML.Data/Transforms/SelectColumnsTransform.cs

```diff
@@ -384,6 +384,27 @@ public static IDataView Create(IHostEnvironment env, ModelLoadContext ctx, IData
             return transform.Transform(input);
         }
 
+        public static IDataTransform CreateKeep(IHostEnvironment env, IDataView input, params string[] keepColumns)
```

Comment:
>IDataTransform CreateKeep [](start = 22, length = 25)

Do you have any place where it requires to be IDataTransform and not IDataView?
If not, I would suggest you to not use arguments class and just create SelectColumnsTransform and pass required parameters into constructor and then call .Transform on input.

Reply:
Sorry Ivan, I misunderstood - so the feedback is to not use Arguments inside the function and return an IDataView instead of IDataTransform. Got it, I will make those changes.

Reply:
I was able to remove the usage of Arguments however I was not able to move to an IDataView. There are other transforms that will use this transform and expect to return an IDataTransform.

---
In reply to: [229034034](https://github.com/dotnet/machinelearning/pull/1371#discussion_r229034034) [](ancestors = 229034034)